src/js/components/SidebarVolumesFilterComponent.jsx

```diff
@@ -0,0 +1,74 @@
+import React from "react/addons";
+
+import AppsStore from "../stores/AppsStore";
+import AppsEvents from "../events/AppsEvents";
+import FilterTypes from "../constants/FilterTypes";
+
+import QueryParamsMixin from "../mixins/QueryParamsMixin";
+
+/**
+ * Health is wealth, peace of mind is happiness.
+ */
+var SidebarVolumesFilterComponent = React.createClass({
+  displayName: "SidebarVolumesFilterComponent",
+
+  mixins: [QueryParamsMixin],
+
+  handleChange: function () {
+    var currentState = this.getQueryParamValue(FilterTypes.VOLUMES) === "true"
+      ? null
+      : true;
+
+    this.setQueryParam(FilterTypes.VOLUMES, currentState);
+  },
+
+  getInitialState: function () {
+    return {
+      appsVolumesCount: ""
+    };
+  },
+
+  componentWillMount: function () {
+    AppsStore.on(AppsEvents.UPDATE_APPS_FILTER_COUNT,
```

Comment:
Gotcha. The `UPDATE_APPS_FILTER_COUNT` event is fired whenever the filter count (health, status, volumes) is updated.  So let's keep it as it is. But please rename the handler otherwise it's kinda confusing.